### PR TITLE
`compression_per_level` should be used for flush and changeable

### PR DESCRIPTION
**Pull request by jay-zhuang**
- Make `compression_per_level` dynamical changeable with `SetOptions`;
- Fix a bug that `compression_per_level` is not used for flush;

Test Plan: CI

### CODE REVIEW

File: options/cf_options.h [hunk 1]
```diff
@@ -271,6 +270,7 @@ struct MutableCFOptions {
   Temperature bottommost_temperature;
 
   uint64_t sample_for_compression;
+  std::vector<CompressionType> compression_per_level;
```
Comment (mrambacher) [inline options/cf_options.h:273]: Minor nit: Can we try to keep the order of the structure elements the same between the ColumnFamilyOptions and the Mutable/Immutable ones?  It makes it easier for comparison/checking purposes.

**Review by mrambacher — APPROVED** [review]
LGTM.

File: options/cf_options.h (hunk 1 above)
Comment (jay-zhuang) [reply options/cf_options.h:273]: `MutableCFOptions` groups the options by memtable, compaction, blob, misc, we're putting it to misc group with other compression options:
https://github.com/facebook/rocksdb/blob/6cb7d64259110345a86ad48252df8dacd8c623cd/options/cf_options.h#L246-L260

**Review by ajkr — APPROVED** [review]
Nice! LGTM

**Pull request closed without merging**